plugins/modules/ipaautomember.py

```diff
@@ -0,0 +1,388 @@
+    if grouping is not None:
+        _args['type'] = to_text(grouping)
+
+    return _args
+
+
+def transform_conditions(conditions):
+    """Transform a list of dicts into a list with the format of key=value."""
+    transformed = []
+    for condition in conditions:
+        transformed.append('='.join(str(x) for x in condition.values()))
+    return transformed
+
+
+def gen_condition_commands(name,
+                           grouping,
+                           module_conditions,
+                           current_conditions):
+    """Return a list of commands to add/remove automember rule conditions."""
+    commands = []
+    add_diff = set(module_conditions) - set(current_conditions)
+    remove_diff = set(current_conditions) - set(module_conditions)
+
+    for item in add_diff:
+        key, condition = item.split("=")
+        condition_args = gen_condition_args(grouping, key, condition, None)
+        commands.append([name, 'automember_add_condition', condition_args])
+
+    for item in remove_diff:
+        key, condition = item.split("=")
+        condition_args = gen_condition_args(grouping, key, condition, None)
+        commands.append([name, 'automember_remove_condition', condition_args])
+
+    return commands
+
+
+def main():
+    ansible_module = AnsibleModule(
+        argument_spec=dict(
+            # general
+            ipaadmin_principal=dict(type="str", default="admin"),
+            ipaadmin_password=dict(type="str", required=False, no_log=True),
+
+            inclusive=dict(type="list", aliases=[
+                           "automemberinclusiveregex"], default=None),
+            exclusive=dict(type="list", aliases=[
+                           "automemberexclusiveregex"], default=None),
+            name=dict(type="list", aliases=["cn"],
+                      default=None, required=True),
+            description=dict(type="str", default=None),
+            type=dict(type='str', required=True,
+                      choices=['group', 'hostgroup']),
+            action=dict(type="str", default="service",
+                        choices=["member", "service"]),
+            state=dict(type="str", default="present",
+                       choices=["present", "absent", "rebuild"]),
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # Get parameters
+
+    # general
+    ipaadmin_principal = ansible_module.params.get("ipaadmin_principal")
+    ipaadmin_password = ansible_module.params.get("ipaadmin_password")
+    names = ansible_module.params.get("name")
+
+    # present
+    description = ansible_module.params.get("description")
+
+    # conditions
+    inclusive = ansible_module.params.get("inclusive")
+    exclusive = ansible_module.params.get("exclusive")
+
+    # action
+    action = ansible_module.params.get("action")
+    # state
+    state = ansible_module.params.get("state")
+
+    # grouping/type
+    grouping = ansible_module.params.get("type")
```

Comment:
Since `type` doesn't work well for variable name, in ansible-freeipa we use `<what>_type`, in this case `grouping_type` or even `automember_type`. Also use this name as attribute in the YAML file. Please, do not use `type`.

Reply:
Done